x/swaprouter/types/listeners.go

```diff
@@ -0,0 +1,21 @@
+package types
+
+import sdk "github.com/cosmos/cosmos-sdk/types"
+
+type PoolCreationListener interface {
+	// AfterPoolCreated is called after CreatePool
+	AfterPoolCreated(ctx sdk.Context, sender sdk.AccAddress, poolId uint64)
+}
+
+type PoolCreationListeners []PoolCreationListener
```

Comment:
Can we name them hooks instead of listeners?

Reply:
According to discussions with @ValarDragon during `twap` implementation, we are going to go towards pub-sub listeners in the future. So it might make sense to start adjusting our abstractions to listeners incrementally